python/cloudtik/core/_private/cluster/cluster_operator.py

```diff
@@ -1144,6 +1144,7 @@ def _set_up_config_for_head_node(config: Dict[str, Any],
     # drop proxy options if they exist, otherwise
     # head node won't be able to connect to workers
     remote_config["auth"].pop("ssh_proxy_command", None)
+    remote_config["auth"].pop("ssh_public_key", None)
```

Comment:
Is there any reason we need to pop ssh_public_key here now?

Reply:
"ssh_public_key" in utils.py will read this ssh_public_key file to get hash for NodeLauncher and in fact, we do not need to use the public key on the head node again.